manifests/container.pp

```diff
@@ -0,0 +1,49 @@
+# @summary Pulp Container plugin
+# @param location_prefix
+#   In the Apache configuration a location with this prefix is exposed. The
+#   version will be appended.
+# @param registry_v1_path
+#   The path beneath the location prefix to forward. This is also appended to
+#   the content base url.
+# @param registry_v2_path
+#   The path beneath the location prefix to forward. This is also appended to
+#   the content base url.
+class foreman_proxy_content::container (
+  String $location_prefix = '/pulpcore_registry',
+  String $registry_v1_path = '/v1/',
+  String $registry_v2_path = '/v2/',
+) {
+
+  $context = {
+    'directories'  => [
+      {
+        'provider'        => 'location',
+        'path'            => "${location_prefix}${$registry_v2_path}",
+        'request_headers' => ["set SSL_CLIENT_S_DN \"admin\""],
+        'requires'        => ["expr %{SSL_CLIENT_S_DN_CN} == \"${facts['networking']['fqdn']}\""]
+      },
+    ],
+    'proxy_pass' => [
+      {
+        'path' => $registry_v1_path,
+        'url'  => "https://127.0.0.1:9090/container_gateway${$registry_v1_path}",
```

Comment:
Please reuse variables here
```suggestion
        'url'  => "${foreman_proxy::real_registered_proxy_url}/container_gateway${registry_v1_path}",
```

The same below. I think this also means you actually can check the peer certificates but you may need to provide the CA to verify the connection. I think that's done via `SSLProxyMachineCertificateFile`. I think the value is `$certs::foreman_proxy::default_ca`.

Reply:
Hm `$certs::foreman_proxy::default_ca` definitely exists, but it's printing out as `Ca[katello-default-ca]`.

Fixing the config so that it's 
```
SSLProxyMachineCertificateFile /etc/pki/katello/certs/katello-default-ca.crt
```
crashes httpd though, so something is missing here anyway.

Reply:
I'll need to take a more detailed look.

Reply:
Removing the `'SSLProxyCheckPeerCN off',  'SSLProxyCheckPeerName off',` bits works actually without any additions, guess I didn't test that before. Going to remove it and we should be all good there.

Reply:
That's because `${foreman_proxy::real_registered_proxy_url}` contains the hostname that's on the certificate. Then the peer check actually passes. `127.0.0.1` is very unlikely to ever be on the certificate.